docs/observability/cloud-trace.md

```diff
@@ -0,0 +1,227 @@
+
+- Trace agent interactions : Cloud Trace continuously gathers and analyzes trace data from your project, enabling you to rapidly diagnose latency issues and errors within your ADK applications. This automatic data collection simplifies the process of identifying problems in complex agent workflows.
+- Debug issues : Quickly diagnose latency issues and errors by analyzing detailed traces. Crucial for understanding issues that manifest as increased communication latency across different services or during specific agent actions like tool calls.
+- In-depth Analysis and Visualization: Trace Explorer is the primary tool for analyzing traces, offering visual aids like heatmaps for span duration and line charts for request/error rates. It also provides a spans table, groupable by service and operation, which gives one-click access to representative traces and a waterfall view to easily identify bottlenecks and sources of errors within your agent's execution path
+
+The following example will assume the following agent directory structure
+
+```
+working_dir/
+├── weather_agent/
+│   ├── agent.py
+│   └── __init__.py
+└── deploy_agent_engine.py
+└── deploy_fast_api_app.py
+└── agent_runner.py
+```
+
+```python
+# weather_agent/agent.py
+
+import os
+from google.adk.agents import Agent
+
+os.environ.setdefault("GOOGLE_CLOUD_PROJECT", "{your-project-id}")
+os.environ.setdefault("GOOGLE_CLOUD_LOCATION", "global")
+os.environ.setdefault("GOOGLE_GENAI_USE_VERTEXAI", "True")
+
+
+# Define a tool function
+def get_weather(city: str) -> dict:
+    """Retrieves the current weather report for a specified city.
+
+    Args:
+        city (str): The name of the city for which to retrieve the weather report.
+
+    Returns:
+        dict: status and result or error msg.
+    """
+    if city.lower() == "new york":
+        return {
+            "status": "success",
+            "report": (
+                "The weather in New York is sunny with a temperature of 25 degrees"
+                " Celsius (77 degrees Fahrenheit)."
+            ),
+        }
+    else:
+        return {
+            "status": "error",
+            "error_message": f"Weather information for '{city}' is not available.",
+        }
+
+
+# Create an agent with tools
+root_agent = Agent(
+    name="weather_agent",
+    model="gemini-2.5-flash",
+    description="Agent to answer questions using weather tools.",
+    instruction="You must use the available tools to find an answer.",
+    tools=[get_weather],
+)
+```
+
+## Cloud Trace Setup
+
+### From Agent Engine Deployment
+
+To enable cloud tracing on ADK Agent Engine deployment, you can add `enable_tracing=True` when initialize the `AdkApp` object
```

Comment:
Rephrase for clarity:
"To enable cloud tracing on ADK Agent Engine deployment"
-->
"To enable cloud tracing for your ADK agent that you are deploying to Agent Engine,"

Reply:
done